### PR TITLE
Dynamic Tags: Fix - When there are no dynamic tags available, clicking on the dynamic switcher shows the promotion dialog

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Dynamic Tags: Fix - When there are no dynamic tags available, clicking on the dynamic switcher shows the promotion dialog even when Elementor Pro is installed and active.

## Quality assurance

- [X] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js [hunk 1]
```diff
@@ -95,6 +95,9 @@ module.exports = Marionette.Behavior.extend( {
 			const proTeaser = Marionette.Renderer.render( '#tmpl-elementor-dynamic-tags-promo' );
 
 			$tagsListInner.append( proTeaser );
+		} else if ( elementor.helpers.hasPro() && elementorPro.config.isActive && ! Object.keys( tags ).length ) {
```
Comment (kobizz) [inline assets/dev/js/editor/components/dynamic-tags/control-behavior.js:98]: Extract `hasPro` and `hasTags` to external variables

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js [hunk 2]
```diff
@@ -95,6 +95,9 @@ module.exports = Marionette.Behavior.extend( {
...
+			// If Elementor Pro is installed and there are no tags to show, don't show an empty popover - notify the user there are no available tags.
+			$tagsListInner.html( '<div class="e-dynamic-tags-empty">' + elementor.translate( 'No matching Dynamic Tags were found.' ) + '</div>' );
```
Comment (kobizz) [inline assets/dev/js/editor/components/dynamic-tags/control-behavior.js:100]: Use the new translation method

File: assets/dev/scss/editor/_dynamic-tags.scss [hunk 3]
```diff
@@ -166,3 +166,10 @@
 		}
 	}
 }
+
+.e-dynamic-tags-empty {
+	padding: 20px 10px 15px;
+	display: block;
```
Comment (kobizz) [inline assets/dev/scss/editor/_dynamic-tags.scss:172]: Do you need this?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/editor/_dynamic-tags.scss (hunk 3 above)
Comment (udidol) [reply assets/dev/scss/editor/_dynamic-tags.scss]: Removed.

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js (hunk 2 above)
Comment (udidol) [reply assets/dev/js/editor/components/dynamic-tags/control-behavior.js]: Done.

File: assets/dev/js/editor/components/dynamic-tags/control-behavior.js (hunk 1 above)
Comment (udidol) [reply assets/dev/js/editor/components/dynamic-tags/control-behavior.js]: Done.

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Comment by arielk** [comment]
Fix: https://github.com/elementor/elementor/pull/12764

**Pull request merged**